### PR TITLE
fix: 2-fa golang snippets

### PR DESCRIPTION
**Pull request by sattvikc**
## Checklist
- [ ] Algolia search needs to be updated? (If there is a new sub docs project, then yes)
- [ ] Sitemap needs to be updated? (If there is a new sub docs project, then yes)
- [ ] Checked for broken links? (Run `cd v2 && MODE=production npx docusaurus build`)
- [ ] Changes required to the demo apps corresponding to the docs?

### CODE REVIEW

File: v2/mfa/backend/second-factor.mdx [hunk 1]
```diff
@@ -785,7 +1071,75 @@ Passwordless.init({
...
+					var userPhoneNumber *string
+					if phoneNumber, ok := session.GetAccessTokenPayloadWithContext(userContext)["phoneNumber"].(string); ok {
+						userPhoneNumber = &phoneNumber
+					}
+					// highlight-end
+
+					if userPhoneNumber != nil {
+						// this means we found a phone number associated to this user.
+						// we will check if the input phone number is the same as this one.
+						if phoneNumber != nil && *phoneNumber != *userPhoneNumber {
```
Comment (rishabhpoddar) [inline v2/mfa/backend/second-factor.mdx:1118]: This condition is wrong (see previous code snippet)

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: v2/mfa/backend/second-factor.mdx [hunk 2]
```diff
@@ -572,7 +625,102 @@ Session.init({
...
+	})
+
+	session.Init(&sessmodels.TypeInput{
+		Override: &sessmodels.OverrideStruct{
+			Functions: func(originalImplementation sessmodels.RecipeInterface) sessmodels.RecipeInterface {
+				oCreateNewSession := *originalImplementation.CreateNewSession
+				/* This function is called after signing in or signing up via the first factor */
+				nCreateNewSession := func(res http.ResponseWriter, userID string, accessTokenPayload map[string]interface{}, sessionData map[string]interface{}, userContext supertokens.UserContext) (sessmodels.SessionContainer, error) {
+					// highlight-start
+					if session, ok := (*userContext)["session"].(*sessmodels.SessionContainer); ok {
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx:705]: image

`session` was a pointer here. Should I de-reference it here itself?

File: v2/mfa/backend/second-factor.mdx [hunk 3]
```diff
@@ -654,7 +802,86 @@ Passwordless.init({
...
+						passwordlessUserInfo, err := passwordless.GetUserByIDWithContext(passwordlessUserId, userContext)
+						if err != nil {
+							return plessmodels.CreateCodePOSTResponse{}, err
+						}
+						userPhoneNumber = passwordlessUserInfo.PhoneNumber
+					}
+					if userPhoneNumber != nil {
+						// this means we found a phone number associated to this user.
+						// we will check if the input phone number is the same as this one.
+						if phoneNumber != nil && *phoneNumber != *userPhoneNumber {
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx]: Fixed

File: v2/mfa/backend/second-factor.mdx [hunk 4]
```diff
@@ -654,7 +802,86 @@ Passwordless.init({
...
+	passwordless.Init(plessmodels.TypeInput{
+		FlowType: "USER_INPUT_CODE",
+		ContactMethodPhone: plessmodels.ContactMethodPhoneConfig{
+			Enabled: true,
+		},
+		Override: &plessmodels.OverrideStruct{
+			APIs: func(originalImplementation plessmodels.APIInterface) plessmodels.APIInterface {
+				/*This API is called to send an OTP*/
+				oCreateCodePOST := *originalImplementation.CreateCodePOST
+				nCreateCodePOST := func(email *string, phoneNumber *string, options plessmodels.APIOptions, userContext supertokens.UserContext) (plessmodels.CreateCodePOSTResponse, error) {
```
[the remark this answers is not in the document]
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx:827]: Fixed

File: v2/mfa/backend/second-factor.mdx (hunk 1 above)
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx]: Fixed

File: v2/mfa/backend/second-factor.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx:828]: Done

File: v2/mfa/backend/second-factor.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (rishabhpoddar) [reply v2/mfa/backend/second-factor.mdx:705]: Yes. You should

**Review by rishabhpoddar — APPROVED** [review]

File: v2/mfa/backend/second-factor.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (sattvikc) [reply v2/mfa/backend/second-factor.mdx]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**